### PR TITLE
Add type annotations and mypy to tests

### PR DESCRIPTION
**Pull request by frenzymadness**
During preparations for my next lesson of the Python and its libraries course, I've decided to add type annotations to the Marshal parser.

### CODE REVIEW

**Comment by frenzymadness** [comment]
by the way, it helped me to uncover some potential issues and anti-patterns so I consider it very useful.

File: marshalparser/marshalparser.py [hunk 1]
```diff
@@ -26,7 +27,7 @@ class Flag_ref:
 
 
 class MarshalParser:
-    def __init__(self, filename):
+    def __init__(self, filename: Path) -> None:
```
Comment (hroncok) [inline marshalparser/marshalparser.py:30]: I'm curious about mypy: Does it encourage you to add `-> None` to class inits?

Comment (frenzymadness) [reply marshalparser/marshalparser.py]: Good catch, I did this one proactively and it's not required for `__init__`.

**Comment by frenzymadness** [comment]
For some reason, `fedora-python-tox:s390x` wasn't a part of the `:latest` manifest. A manual restart of the cron job for fedora-python-tox fixed it. I'll try to find a way how to test the deployed manifest.

**Pull request merged**